sklearn/utils/validation.py

```diff
@@ -361,11 +361,14 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         array = _ensure_sparse_format(array, accept_sparse, dtype, copy,
                                       force_all_finite)
     else:
+        array = np.array(array, dtype=dtype, order=order, copy=copy)
+
         if ensure_2d:
-            array = np.atleast_2d(array)
+            if array.ndim == 1:
+                warnings.warn("Array needs atleast 2 dimnesions",
```

Comment:
This needs a better message, like "Passing 1d arrays as data is deprecated and will be removed in 0.18. Reshape your data either using ``X.reshape(-1, 1)`` if your data has a single feature or ``X.reshape(1, -1)`` if it contains a single sample."